msal/build.gradle

```diff
@@ -277,7 +277,7 @@ publishing {
 }
 
 tasks.whenTaskAdded { task ->
-    if (task.name == 'assembleDebug' || task.name == 'assembleRelease') {
+    if (task.name == 'assembleDebug' || task.name == 'assembleRelease' || task.name == 'assembleDistRelease') {
```

Comment:
What about the other tasks?

Reply:
There are more assemble tasks - let's run for all of them

Reply:
Good point, that way we catch `lint` and other errors early

Reply:
These are our current `assemble` variants -- I don't think we necessarily want to enable this for `assembleSnapshot` (as it will slow that build down, generate unused artifacts). We could do `task.name.contains('assemble') && !task.name.equals('assembleSnapshot')`.

Thoughts? @shahzaibj

```
brianmel@brianmel-devbox MINGW64 /c/dev/android-complete/msal (iambmelt/simple-build-prep)
$ ./gradlew tasks | egrep assemble
assemble - Assemble main outputs for all the variants.
assembleAndroidTest - Assembles all the Test applications.
assembleDebug - Assembles main outputs for all Debug variants.
assembleDist - Assembles main outputs for all Dist variants.
assembleLocal - Assembles main outputs for all Local variants.
assembleRelease - Assembles main outputs for all Release variants.
assembleSnapshot - Assembles main outputs for all Snapshot variants.

```

Reply:
I think there are even more tasks actually:

```
- assemble
- assembleLocal
- assembleLocalDebug
- assembleLocalRelease
- assembleSnapshot
- assembleSnapshotDebug
- assembleSnapshotRelease
- assembleDist
- assembleDistDebug
- assembleDistRelease 
```
There's actually even more tasks for testing (`assembleLocalReleaseUnitTest` and many more)....I didn't list them here.

If we want to avoid running for snapshot, then I think we want to avoid for each build type of snapshot (debug vs release vs task for assemble test).

May be we can do this:

`if (task.name.contains('assemble') && !task.name.contains('Snapshot'))`

but most likely we also want to avoid the test tasks so then, we can do like:

`if (task.name.contains('assemble') && !task.name.contains('Snapshot') && !task.name.contains('Test'))`

I'd leave up to you to decide how want to do it @iambmelt .

Also if @kreedula has any input.

Reply:
See 653bf3835853ae48449895940f772c68164a4856